### PR TITLE
Configure identity provider for a components through the Data View

### PR DESCRIPTION
**Pull request by mshabarov**
Depends on https://github.com/vaadin/flow/pull/8581

### CODE REVIEW

File: vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridDataViewImpl.java [hunk 1]
```diff
@@ -69,4 +70,10 @@ public T getItemOnRow(int rowIndex) {
         }
         return getItems().skip(rowIndex).findFirst().orElse(null);
     }
+
+    @Override
+    public void setIdentifierProvider(IdentifierProvider<T> identifierProvider) {
```
Comment (pleku) [inline vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridDataViewImpl.java:75]: This should also be added to `GridListDataView` ? 

(I will need to add it to `GridLazyDataView` too)

File: vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridDataViewImpl.java [hunk 2]
```diff
@@ -69,4 +70,10 @@ public T getItemOnRow(int rowIndex) {
...
+        super.setIdentifierProvider(identifierProvider);
+        dataCommunicator.getKeyMapper().setIdentifierGetter(identifierProvider);
```
[the remark this answers is not in the document]
Comment (mshabarov) [reply vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridDataViewImpl.java:77]: Done

File: vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridDataViewImpl.java (hunk 1 above)
Comment (mshabarov) [reply vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridDataViewImpl.java:75]: Done

File: vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridListDataView.java [hunk 3]
```diff
@@ -57,4 +58,9 @@ public int getSize() {
         return dataCommunicator.getDataSize();
     }
 
+    @Override
+    public void setIdentifierProvider(IdentifierProvider<T> identifierProvider) {
```
Comment (pleku) [inline vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridListDataView.java:62]: Now this method is not used/tested from anywhere, making it possible to remove it without any tests to fail

Comment (mshabarov) [reply vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridListDataView.java:62]: It is tested in `GridDataViewImplTest`, however the code is masked a bit: 
```
        // In-memory grid data view
        dataView = component.getListDataView();
```
But it should be in the `GridListDataViewTest` apparently.

Comment (pleku) [reply vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/dataview/GridListDataView.java:62]: Well if it is tested and cannot be removed accidentally then it is enough

**Pull request merged**